### DISCUSSION TITLE
Improving the federation with fidelity bonds

### DISCUSSION THREAD

**Discussion by chris-belcher** [fedimint/fedimint#1446] [General]
I recently watched a talk about this project on the open source stage at the miami bitcoin conference.

I got the idea that you might benefit by using fidelity bonds to choose the people who hold the keys of the server's multisig. I think it fits very well with the cypherpunk ethos of the project.

https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020489.html

Let me know if you have any questions, I'm happy to elaborate.

**Comment by dpc** [comment]
I don't think  think this require any explicit support in software. I also kind of doubt it will be very popular in a general case. As a user I would rather entrust my custodial checking account with some geographically dispersed federation including known and reputable regulated institutions, and the fact that some anon locked some coins does not make me feel all that much better about trusting them.

**Comment by elsirion** [comment]
I think there will be different use cases for federated e-cash:
* local community custody: no deed for fidelity bonds, you know the people
* global, nym-run federations: might add some value
* global, institutional federations run out of friendly jurisdictions: questionable value add

Is there a standard for such fidelity bods? Ideally there would be a common protocol that allows setting up bonds for different services/consumers. I.e. `bond-tool <service-identifier> <amount> <time>` that would take a (unique) service identifier like `minimint:<federation-cfg-hash>` such that fidelity bonds can not be reused. This command would generate an address to send coins to and a proof. The address could be a TR address using one tapleaf to commit to the service identifier and another to encode the time lock. Or just use key tweaking and any address scheme to commit to the identifier.

**Comment by ghost** [comment]
There is a proposed BIP to standardize fidelity bonds, AFAIK the same as the current join market integration.
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020389.html

**Comment by chris-belcher** [comment]
Remember that reputation has downsides too. Namely exit scams. Because when someone builds up a good reputation it's not easy to convert that into something else, the only way to convert is to exit scam (lose your reputation but gain all the money in custody). So in systems that depend on reputation we often see the ecosystem is full of exit scams.

Reputation is also centralizing, as it's very hard for a new custodian to build up a reputation to compete with the existing custodial who have a long-running good reputation. Anyone sane will go with the big already-existing actors with good reputations (until they decide to disappear with the money, and because its so centralized there's a lot of).

This isn't to say that you can't build a good system based on reputation, especially if you have some mechanism to sell reputation without exit scamming, but it's worth knowing about other options too.

**Comment by elsirion** [comment]
I think that's actually the biggest power of federation: sowing dissent over when to exit-scam. If one party proposes too early while others don't want to they will be removed and hopefully replaced by someone with lower time preference. This should keep them from proposing in the first place. @frankbraun and @JonathanLogan had some interesting thoughts on that in their [SCRIT whitepaper](https://github.com/scritcash/scrit-whitepaper/blob/master/scrit-whitepaper.pdf) in section 12, in particular Cartel Theory.

**Comment by dpc** [comment]
I generally don't see fidelity bonds as practical replacement for anything.

"You locked 100BTC for 2 years, how is that exactly making your exit scam harder?" The amounts of funds that would had to be locked to be a meaningful barrier to prevent multiple keys in the federation controlled by the same actor are not practical. If you can lock 100BTC for 2 years, chances are you can lock 300 BTC for 2 years, control 3 keys in the 4 keys federation and reliably exit scam the federation at will.

And increasing the bond is barrier mostly only for honest actors. Since bonds have an opportunity cost (one could lock these funds in a LN, or lend them out for a %), the higher you go, the more costly it becomes. And why would you lock so much, if you are *not* planning to profit disproportionately by eventually exit scamming?

**Comment by dpc** [comment]
It makes more sense to trust a mint consisting of actors that are running things as cheaply as possible, in the hopes that securing the income stream of the service itself in the long term is in their best interest, and colluding to exit scam is just not in their own self-interest.

If the mint takes a 1% cut on every transfer (quite modest, comparing to e.g. CC fees), and the velocity is high enough - lets say all funds change hands around 10 times every year, then the mint is making 10% of all the funds they hold every year. So an exit scam is worth 10 years of actually operating (assuming no feature growth). Seems like exit scam is somewhat appealing.

By increasing the turnover rate (less funds, but moving much faster, thus generating more fees), we make the exist scam less likely. But this has diminishing returns, because no one is going to use a system that basically burns their money extra fast with fees.

Increasing the number of stakeholders in the federation, especially by actually legal or at least publicly known entities, seems to have the best reliability multiplier. But it might be difficult from legal perspective, until legal systems support this use case.

The fidelity bonds would make more sense if they were smaller, but instead would be held in a (possibly also federated) custody, so there is actually something at stake. Possibly the mint federation could be anonymous, but hold their stakes in known, legal, reputable entities. This way (maybe, hopefuly?) legal entities can be isolated from anonymous ones, and the fidelity bonds can be smaller (more economical) because they actually carry a loss with them, and are there not to try to make sibil attack harder, but to tip the scale of the exist scam vs income self-interest calculation.

**Comment by NicolaLS** [comment]
@dpc I mean it doesn't make the exit scam harder directly,  that's what the federation is for, but it's purpose is to insure that the federation is actually composed of different entities. So you could say it implies 'harder exit scams'. Though I agree with your other point in that this scheme could be counter productive for such things like minimint ( invite scammers, excluding good actors). Imo. it can only be helpful when there are atomic/natural incentives (like competing for money in PoW) for doing something and not abstract incentive structures (like ideology, etc..). To your last point: I don't think that it will be possible for legal entities to work with chaumian banks, nor would it be in the chaumian banks interest to interact with legal entities.

**Comment by dpc** [comment]
> I don't think that it will be possible for legal entities to work with chaumian banks, nor would it be in the chaumian banks interest to interact with legal entities.

Within the current legal frameworks present in the most of the world, that is correct. It does not have to be the case forever, though.

After the collapse of the current financial system and inevitable hyperbitcoinization, situation might drastically change. As states become cut off from the control of money, they will need to reconsider their interests and role, become leaner and up their competitive game as capital flows are harder to block.

All it takes is a handful of countries allowing legal entities to operate chaumian mints (e.g. in federation), and all of the world can use them. You can just include handful of actually known entities in the federation, some other federation member can stay anonymous, and chances of collusion drop to zero fast.

Without such reputable entities, I find the usefulness of chaumian mints greatly limited. The napkin economic calculation doesn't look great. Why wait 10 years (10%/year yield) if you can disappear with all the funds today, and start from scratch under new identity? Reputation doesn't really work for anonymous identities, bonds don't help, AFAICT. In practice people tend to be irrationally honest (just look at all this open source code that barely anyone reviews, yet it isn't malicious 99.999% of the time), so in practice such anonymous federated mints might actually operate, but it's hard to put a lot of trust in them, when you don't know the operators and know that game theory is against you.

**Comment by BitcoinBrief** [comment]
What if fidelity bonds aren't used to create trust in the guardians, but to facilitate enforceability of civil judgments between members who custody their bitcoin with the federation?  This would be in the community use-case, where residents of a hyperbitcoinized area (ideally free autonomous economic zones or cities) agree to deposit a percentage of their funds in fidelity bonds upon taking up residence.  If adjudicated liable for a breach of the community's social contract (which would be an actual written agreement each resident assets to), a portion of the liable member's fidelity bond would be forfeited and transferred to compensate the injured party.  

This solves for the problem of enforcing civil judgments against bitcoin owners (impossible without coercion if self-custodied privately) to peacefully resolve disputes.  

I'm a lawyer currently researching this topic for a peer-reviewed article to be published by the Texas Bitcoin Foundation.  I think fedimint could provide a workable platform for scaling bitcoin-based dispute resolution systems to the community level.  I would love any feedback folks here could provide.  I'm attaching my article abstract here.
[Satoshi Papers abstract.docx](https://github.com/fedimint/fedimint/files/9792454/Satoshi.Papers.abstract.docx)

**Comment by dpc** [comment]
> What if fidelity bonds aren't used to create trust in the guardians, but to facilitate enforceability of civil judgments between members who custody their bitcoin with the federation?

An interesting idea, though tangential to the spirit of the idea from the original post. I forwarded it around. Thought I guess it might not be compatible with the "mint" part of Fedimint (issuing anonymous notes), and only with the N-of-M custody.
